packages/widgets/src/dns-hole/controls/component.tsx

```diff
@@ -0,0 +1,133 @@
+"use client";
+
+import { useState } from "react";
+import type { BoxProps } from "@mantine/core";
+import { ActionIcon, Badge, Box, Button, Card, Flex, Group, Stack, Text, Tooltip, UnstyledButton } from "@mantine/core";
+import { useDisclosure } from "@mantine/hooks";
+import { IconClockPause, IconPlayerPlay, IconPlayerStop } from "@tabler/icons-react";
+
+import { clientApi } from "@homarr/api/client";
+import { useI18n } from "@homarr/translation/client";
+
+import type { WidgetComponentProps, WidgetProps } from "../../definition";
+import { NoIntegrationSelectedError } from "../../errors";
+import TimerModal from "./TimerModal";
+
+const dnsLightStatus = (currentStatus: string): "blue" | "green" | "red" => {
+  if (currentStatus === "enabled") {
+    return "green";
+  }
+  return "red";
+};
+
+export default function DnsHoleControlsWidget({ options, integrationIds }: WidgetComponentProps<"dnsHoleControls">) {
+  const integrationId = integrationIds.at(0);
```

Comment:
This will limit the widget to only one integration. The old one supports multiple. You need to support that too. It will be a bit harder to implement though. See the other widgets how we implemented it there.

Reply:
Need some help with this